kernels/test/op_mul_test.cpp

```diff
@@ -472,6 +514,19 @@ TEST_F(OpMulOutTest, BothScalarInputBroadcastTest) {
   test_both_scalar_input_broadcast<ScalarType::BFloat16>();
 }
 
+TEST_F(OpMulOutTest, AllComplexDtypesSupported) {
+#define TEST_ENTRY(ctype, dtype) test_complex_dtype<ctype, ScalarType::dtype>();
+  if (torch::executor::testing::SupportedFeatures::get()->is_aten) {
+    ET_FORALL_COMPLEX_TYPES(TEST_ENTRY);
+  } else {
+    // Disabling this for portable mode for now because currently we only support
+    // complex dtypes in optimized mode. Portable mode needs support in elementwise_ops.h
+    // for complex dtypes, which we don't have yet.
+    //ET_FORALL_COMPLEXH_TYPES(TEST_ENTRY);
```

Comment:
The fact that you are commenting out this line means none of the changes you added in this PR are being tested. I agree you shouldn't be blocked by adding support in the elementwise macros. You have two possible solutions:
a) Add your optimized-only tests under https://github.com/pytorch/executorch/tree/main/kernels/optimized/test
b) Enable complex in portable op_mul without messing around with the elementise macros, just add an if/else and add similar code to the one you added in the optimized op.
I think b) will be way easier

Reply:
@manuelcandales took approach b and updated the PR.